### PR TITLE
[DOC, REF] Refactor select_comps and add shape checks to several functions

### PR DESCRIPTION
**Pull request by tsalo**
Changes proposed in this pull request:

- Rename input arguments for `tedana.selection._utils.getelbow_*` functions from `ks` (which I assume referred to kappa values) and `val` to `arr` and `return_val`, because these seem more descriptive.
- Rename `tedana.model.fit.fitmodels_direct` input argument `t2sG` to `t2s_full`. ~~Also change default value for argument `fout` to `False` from `None`, although this argument is unused and could be dropped entirely.~~ Also drop parameter `fout`.
- Add argument shape/size checks to `tedana.model.fit.fitmodels_direct`, `tedana.model.fit.computefeats2`, `tedana.model.fit.get_coeffs`, and `tedana.model.fit.gscontrol_raw`.
    - ~~These simple `assert`-based checks can be replaced with something more elegant, but I think they're important for anyone who uses these functions themselves (i.e., not when they're called within the workflows). Also, they caught some things that helped me correct the specified shapes of the inputs in the docstrings.~~ Use ValueErrors to check shapes and sizes.
- Drop argument `start_echo` from `tedana.model.monoexponential.fit_decay` and `tedana.model.monoexponential.fit_decay_ts`. I don't think there's much value in keeping this argument, and it just seemed to confuse everything, but I can add it back if necessary. Also, it seemed like the code wasn't really set up to use `start_echo` properly.
- Refactor the `t2ss`-generating for loop in `fit_decay` to make it easier to understand (IMO). I can revert this change if others don't feel it's easier to understand.
- Update `tedana.tests.test_tedana` to output a list of the failing volumes instead of raising an Exception on the first one. It might make it easier for us to detect the source of breaking changes.
- Refactor `tedana.selection.select_comps.selcomps`. None of these changes should step on @handwerkerd's comments, but please let me know if they do.
    - Add argument shape checks. 
    - Change some uninformative variable names:
        - `nc` --> `all_comps`
        - `ncl` --> `acc_comps`: This is an array of all component numbers that is whittled down throughout the selection process to leave just the accepted component numbers.
        - Iterator `ii` --> `comp_num` when looping through component numbers.
        - Log of variance explained (`varex_`) --> `varex_log`
    - Replace `Xz = (X - X.mean()) / X.std()` pattern (which is mixed in apparently randomly with `zscore` usage) with `Xz = stats.zscore(X)`.

### CODE REVIEW

**Comment by emdupre** [comment]
@handwerkerd I know that `selcomps` is being discussed generally in #97, but it'd be great to also get your look-through here !

File: tedana/model/fit.py [hunk 1]
```diff
@@ -42,8 +46,9 @@ def fitmodels_direct(catd, mmix, mask, t2s, t2sG, tes, combmode, ref_img,
         Poser 2006
     ref_img : str or img_like
         Reference image to dictate how outputs are saved to disk
-    fout : bool
-        Whether to output per-component TE-dependence maps. Default: None
+    fout : bool, optional
```
[the remark this answers is not in the document]
Comment (tsalo) [reply tedana/model/fit.py]: 👍

File: tedana/model/fit.py [hunk 2]
```diff
@@ -62,9 +67,15 @@ def fitmodels_direct(catd, mmix, mask, t2s, t2sG, tes, combmode, ref_img,
     betas : :obj:`numpy.ndarray`
     mmix_new : :obj:`numpy.ndarray`
     """
+    assert catd.shape[0] == t2s.shape[0] == mask.shape[0]
```
[the remark this answers is not in the document]
Comment (tsalo) [reply tedana/model/fit.py]: Absolutely. I'll replace the assertions with ValueErrors.

**Comment by emdupre** [comment]
A small follow-up: our coverage is going down with this PR, but I think we could easily bump it back up by checking that the `ValueError`'s are actually raised. Any chance you could add (at least a few of) those in ?

**Comment by emdupre** [comment]
This LGTM -- I'll leave it open for another day so @handwerkerd (or anyone else ! @rmarkello ?) can comment if they'd like to !

**Comment by handwerkerd** [comment]
I don't have time to look through this in depth in the next couple of days. Just skimming, I could quibble here or there, but things look fairly benign. If you want to wait a few days, I might be able to look at this more carefully, but I'm fine if you just merge.

**Comment by emdupre** [comment]
Happy to wait until the weekend to give more time to the gitter discussion ! @tsalo and I will also be at INCF over the next few days, so we probably won't have too much time to work on this until the weekend.

If you're just too busy though, no worries ! There's still quite a few steps to get feedback and iterate on at this point, so I'm sure most of this code will come up for review again at some point :smile:

**Comment by emdupre** [comment]
Ok, merging ! Thanks again, @tsalo !

**Pull request merged**